src/components/NewWorkspaceModal.js

```diff
@@ -111,7 +111,12 @@ const NewWorkspaceModal = withDisplayName('NewWorkspaceModal', ({
         }],
         async () => {
           const workspace = await Ajax().Workspaces.create(body)
-          Ajax().Metrics.captureEvent(Events.workspaceCreate, extractWorkspaceDetails(
+          // Only invoke Leo if we are within an Azure Workspace
+          if (!workspace.googleProject) {
```

Comment:
I couldn't use `isAzureWorkspace` in this context, since the payload returned from `Ajax().Workspaces.create(body)` doesn't contain the `cloudPlatform` key.

I could update the function of `isAzureWorkspace`, but unfortunately as is, `!workspace.googleProject` was the only way I could understand if it was Azure. Let me know your thoughts here!

Reply:
the Rawls swagger docs say that `cloudPlatform` is available, though you get `{"cloudPlatform":"whatever"}` instead of the `{"workspace": {"cloudPlatform": "whatever"}}` that `isAzureWorkspace` expects. Just tossing this out there; I don't know if it is worth refactoring `isAzureWorkspace` to support this.

Reply:
Here is an example payload that is returned when calling `https://rawls.dsde-dev.broadinstitute.org/api/workspaces`

```
{"attributes":{"description":"aaron-new-test"},"authorizationDomain":[],"bucketName":"","createdBy":"aaronkanzer@gmail.com","createdDate":"2023-01-18T19:44:58.243Z","googleProject":"","isLocked":false,"lastModified":"2023-01-18T19:44:58.243Z","name":"new-test","namespace":"ajSharedBilling2023","workspaceId":"e941c71e-9cb0-4d9d-9649-7e91ef9fe25a","workspaceType":"mc","workspaceVersion":"v2"}
```

Could we include `!workspace.googleProject` as an `||` logic in `isAzureWorkspace`?

Reply:
ah interesting! So, the Rawls swagger definition is just plain wrong then - blech.

Reply:
Ugh, I forgot that create and clone workspace don't include `cloudPlatform` (which I actually commented on at line 122!). I will create a Workspaces ticket for us to add that to the response. In the meantime, define a variable with the result of `getProjectCloudPlatform()`, and use the variable here and at line 122.

Reply:
Actually, you can use `isAzureBillingProject` here, and leave line 122 alone.

Reply:
Updated!